api/README.md

```diff
@@ -0,0 +1,46 @@
+# ETL API
+
+API for ETL lets you edit metadata for indicators by creating `*.meta.override.yml` file which overrides
+YAML file at a grapher level. With this file the API then executes:
+
+1. Slow track - commit and push to ETL which fully rebuilds the step (> minutes)
+2. Fast track - run `etl step --grapher` to update MySQL and R2 file (typically < 1 minute)
+2. Super Fast track - update indicator R2 file so that Admin shows updated values (enabling live preview of a chart or data page)
+
+Since Admin & our site is powered by R2 JSON files, we don't have to deploy the site.
+
+## Limitations
+
+It doesn't work for old non-ETL datasets and older steps that don't use `create_dataset` function. If user tries to use it for those datasets, API returns 4xx errors (together with other validation errors).
+
+## Observability
+
+If env `SLACK_API_TOKEN` is set, it'll send all requests & responses to a slack channel (with highlighted warnings). There's also a bugsnag integration for error monitoring.
+
+## Instructions
+
+1. Run `make api` to start ETL API on http://localhost:8000/v1/indicators
+2. Enable "Slow track" by setting `ETL_API_COMMIT=1` which commits the changed file and pushes it to origin.
+
+## Sample request
+
+The following request edits indicator with id `540251`. It sets decimal places, changes its title and
+triggers ETL rebuild (without committing the code).
+
+```
+echo '{
+  "indicator": {
+    "name": "My new name",
+    "display": {
+      "numDecimalPlaces": 2
+    }
+  },
+  "dataApiUrl": "https://api-staging.owid.io/mojmir/v1/indicators/",
+  "triggerETL": true,
+  "dryRun": false
+}' | http PUT http://127.0.0.1:8000/v1/indicators/540251
+
+```
+
+`dataApiUrl` points to the Data API that is used by Admin. If ETL API uses different Data API, it'll
+raise an error (to prevent confusion when querying the wrong ETL API).
```

Comment:
At first I didn't understand this, but if I've understood you right, this is to avoid any misunderstandings/misconfiguration on the Grapher side. Grapher expects to be making the change in e.g. prod, then we don't want to lie to grapher and only make the change in staging.

That seems sensible, though I'll give feedback on it when reviewing the other PR to understand how things are configured there.

Reply:
Yeah, exactly. I've spent tons of time debugging something just to find out my env file is pointing to a different environment. The fact that we're doing fallbacks to producion in Data API makes it even more confusing.